packages/opencensus-web-instrumentation-zone/src/xhr-interceptor.ts

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright 2019, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AsyncTask, XHRWithUrl } from './zone-types';
+import {
+  RootSpan,
+  Span,
+  ATTRIBUTE_HTTP_STATUS_CODE,
+  ATTRIBUTE_HTTP_METHOD,
+  parseUrl,
+  SpanKind,
+} from '@opencensus/web-core';
+import { getXhrPerfomanceData } from './perf-resource-timing-selector';
+import { traceOriginMatchesOrSameOrigin, isTrackedTask } from './util';
+import { spanContextToTraceParent } from '@opencensus/web-propagation-tracecontext';
+import {
+  annotationsForPerfTimeFields,
+  PerformanceResourceTimingExtended,
+  PERFORMANCE_ENTRY_EVENTS,
+  getResourceSpan,
+} from '@opencensus/web-instrumentation-perf';
+
+// Map intended to keep track of current XHR objects
+// associated to a span.
+const xhrSpans = new Map<XHRWithUrl, Span>();
```

Comment:
Nit: should this be named as `XhrWithUrl` to match the Google casing conventions?